src/main/java/org/spongepowered/api/command/args/PatternMatchingCommandElement.java

```diff
@@ -87,6 +89,18 @@ Pattern getFormattedPattern(String input) {
     }
 
     /**
+     * Tests a string against a set of valid choices to see if it is a
+     * case-insensitive match.
+     *
+     * @param choices The choices available to match against
+     * @param potentialChoice The potential choice
+     * @return If matched, an {@link Optional} containing the matched value
+     */
+    protected Optional<Object> getExactMatch(final Iterable<String> choices, final String potentialChoice) {
+        return Iterables.tryFind(choices, potentialChoice::equalsIgnoreCase).toJavaUtil().map(this::getValue);
+    }
```

Comment:
This forces case insensitivity for *anything* that uses patternmatchingcommandelement, I understand that it might be wanted for players / users, but I think this is overstepping somewhat, at least until 1.13

Reply:
It’s not - I’m mirroring what the original code intended, and it tries to do a case insensitivity check for exact matches now, just in the wrong place.